lib/tasks/posts.rake

```diff
@@ -42,31 +42,26 @@ end
 
 desc 'Rebake all posts matching string/regex and optionally delay the loop'
 task 'posts:rebake_match', [:pattern, :type, :delay] => [:environment] do |_, args|
+  args.with_defaults(type: 'string')
   pattern = args[:pattern]
-  type = args[:type]
-  type = type.downcase if type
-  delay = args[:delay].to_i if args[:delay]
+  type = args[:type]&.downcase
+  delay = args[:delay]&.to_i
+
   if !pattern
     puts "ERROR: Expecting rake posts:rebake_match[pattern,type,delay]"
     exit 1
   elsif delay && delay < 1
     puts "ERROR: delay parameter should be an integer and greater than 0"
     exit 1
-  end
-
-  if type == "regex"
-    search = Post.where("raw ~ ?", pattern)
-  elsif type == "string" || !type
-    search = Post.where("raw ILIKE ?", "%#{pattern}%")
-  else
+  elsif type != 'string' && type != 'regex'
     puts "ERROR: Expecting rake posts:rebake_match[pattern,type] where type is string or regex"
     exit 1
   end
 
   rebaked = 0
   total = search.count
 
-  search.find_each do |post|
```

Comment:
Hmmm we stopped using find_each here so every single post gets loaded into memory, this is not really going to work on large databases.

Reply:
Sorry I'm a bit lost here; this _is_ using `find_each` so not all records are loaded at once. Are you referring to the other tasks that use `each`?

Reply:
see: https://apidock.com/rails/ActiveRecord/Batches/ClassMethods/find_each find_each performs the work in batchs so memory usage is low.

Reply:
ahh did not notice you did a find each after the raw_match, my bad